scipy/interpolate/_interpolate.py

```diff
@@ -385,6 +385,8 @@ class interp1d(_Interpolator1D):
     """
     Interpolate a 1-D function.
 
+    .. legacy:: function
```

Comment:
Nitpick, but this is a class. From the code above, I am not sure if it works with a class. It should right?

Reply:
It should! Updated it, thanks